service/history/api/respondworkflowtaskcompleted/api.go

```diff
@@ -416,8 +419,14 @@ func (handler *WorkflowTaskCompletedHandler) Invoke(
 			// Flush buffer event before terminating the workflow
 			ms.FlushBufferedEvents()
 
-			if err := workflow.TerminateWorkflow(ms, wtFailedCause.Message(), nil,
-				consts.IdentityHistoryService, false); err != nil {
+			_, err := ms.AddWorkflowExecutionTerminatedEvent(
```

Comment:
Please check if there's any existing functional tests exercising this code path.